### PR TITLE
Major Refactor to Recipe Views

### PR DESCRIPTION
**Pull request by narenrameshb**
I refactored my entire process stage to adhere fully to clean architecture. I also implemented a RecipeViewFactory and made the necessary changes to the ViewManager.

### CODE REVIEW

**Review by mfatir-y — APPROVED** [review]
All files follow clean architecture. I do not spot any issues either. Perfectly done! I would advise adding javadoc to the tests too so they are easier to understand to someone not familiar with your code.

File: src/main/java/com/intelliChef/view/ViewManager.java [hunk 1]
```diff
@@ -4,21 +4,39 @@
...
+    private RecipeListView currentRecipeListView;
+    private RecipeDetailView currentRecipeDetailView;
 
-
-    public ViewManager(ViewFactory viewFactory, AnalyzeImageInteractor analyzeImageInteractor, RecipeViewFactory recipeviewFactory) {
+    public ViewManager(
+            ViewFactory viewFactory,
+            AnalyzeImageInteractor analyzeImageInteractor,
+            RecipeViewFactory recipeViewFactory,
+            LoadRecipesUseCase loadRecipesUseCase,
```
Comment (EnzaDinza) [inline src/main/java/com/intelliChef/view/ViewManager.java:31]: Where are these initialized? Is it necessary for them to be passed as parameters here?

File: src/main/java/com/intelliChef/Main.java
Comment (EnzaDinza) [inline src/main/java/com/intelliChef/Main.java:1]: Could this object creation logic be moved to your factory to minimize the logic in the main?

**Review by EnzaDinza — APPROVED** [review]
Seems to have no major dependency problem. Added some comments in the code.

**Pull request merged**